src/controllers/subjectController.ts

```diff
@@ -0,0 +1,79 @@
+import { Request, Response } from 'express';
+import mongoose from 'mongoose'; 
+import { isNil, isEmpty, some } from 'lodash';
+
+import * as subjectService from '@services/subjectService';
+import { HTTP_STATUS } from '@utils/constants';
+import { MESSAGES } from '@utils/message';
+import { ISubject } from '@models/Subject';
+import { sendResponse } from '@utils/Respons/response';
+
+export const createSubject = async (req: Request, res: Response) => {
+  try {
+    const subject : ISubject | null = await subjectService.createSubject(req.body);
+    sendResponse(res, HTTP_STATUS.CREATED, MESSAGES.RECORD_ADDED_SUCCESSFULLY, subject);
+  } catch (error) {
+    if (error instanceof mongoose.Error.ValidationError) {
+      const errors = Object.values(error.errors).map((err) => err.message);
+      return sendResponse(res, HTTP_STATUS.BAD_REQUEST, errors);
+    } else {
+      return sendResponse(res, HTTP_STATUS.INTERNAL_SERVER_ERROR, MESSAGES.INTERNAL_SERVER_ERROR);
+    }
+  }
+};
```

Comment:
can we avoid this if else statement everywhere? by creating `sendErrorResponse` function or by any other approach?